tests/e2e/e2e_k8s_version_test.go

```diff
@@ -0,0 +1,121 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/openservicemesh/osm/tests/framework"
+)
+
+// This test is meant to install k8s clusters of different versions and run
+// a correctness test to ensure OSM works on these versions.
+var _ = OSMDescribe("Test HTTP traffic for different k8s versions",
+	OSMDescribeInfo{
+		Tier:   2,
+		Bucket: 1,
+	},
+	func() {
+		Context("Version v1.19.7", func() {
+			testK8sVersion("v1.19.7")
+		})
+		Context("Version v1.18.15", func() {
+			testK8sVersion("v1.18.15")
+		})
+		Context("Version v1.17.17", func() {
+			testK8sVersion("v1.17.17")
+		})
+	})
+
+func testK8sVersion(version string) {
+	const sourceName = "client"
+	const destName = "server"
+	var ns = []string{sourceName, destName}
+
+	Td.ClusterVersion = version // set the cluster version to test
+
+	It("Tests HTTP traffic for client pod -> server pod", func() {
+		if Td.InstType != KindCluster {
```

Comment:
~move to line 21 before all contexts are exec to avoid entering It~ nvm, no InstType at that stage